zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/DataStoreManager.scala

```diff
@@ -233,10 +274,24 @@ object DataStoreManager {
 
   case class AskInfo(who: String)
 
-  case class Register(dataset: String, schema: Schema)
+  case class Register(dataset: String, schema: UnresolvedSchema)
+
+  object Register{
+    implicit val registerReader: Reads[Register] = {
```

Comment:
To validate the json request sent by user in `Application.register` and `Application.deregister`.